lib/activerecord/all/activerecord.rbi

```diff
@@ -1,36 +1,775 @@
+      foreign_key: T.nilable(T.any(Symbol, String)),
+      foreign_type: T.nilable(T.any(Symbol, String)),
+      inverse_of: T.nilable(T.any(Symbol, String)),
+      optional: T.nilable(T::Boolean),
+      polymorphic: T.nilable(T.any(Symbol, String)),
+      primary_key: T.nilable(T.any(Symbol, String)),
+      required: T.nilable(T::Boolean),
+      touch: T.nilable(T::Boolean),
+      validate: T.nilable(T::Boolean),
+      default: T.nilable(T.proc.returns(T.untyped))
+    ).void
+  end
+  def belongs_to(
+    name,
+    scope = nil,
+    autosave: nil,
+    class_name: nil,
+    counter_cache: nil,
+    dependent: nil,
+    foreign_key: nil,
+    foreign_type: nil,
+    inverse_of: nil,
+    optional: nil,
     polymorphic: nil,
+    primary_key: nil,
+    required: nil,
     touch: nil,
     validate: nil,
-    optional: nil
+    default: nil
+  ); end
+
+  sig do
+    params(
+      name: T.nilable(T.any(Symbol, String)),
+      scope: T.nilable(T.proc.void),
+      after_add: T.nilable(T.any(Symbol, String, T.proc.void)),
+      after_remove: T.nilable(T.any(Symbol, String, T.proc.void)),
+      association_foreign_key: T.nilable(T.any(Symbol, String)),
+      autosave: T.nilable(T::Boolean),
+      before_add: T.nilable(T.any(Symbol, String, T.proc.void)),
+      before_remove: T.nilable(T.any(Symbol, String, T.proc.void)),
+      blk: T.nilable(T.proc.void),
+      class_name: T.nilable(T.any(Symbol, String)),
+      extend: T.nilable(T.any(Module, T::Array[Module])),
+      foreign_key: T.nilable(T.any(Symbol, String)),
+      join_table: T.nilable(T.any(Symbol, String)),
+      validate: T.nilable(T::Boolean),
+    ).void
+  end
+  def has_and_belongs_to_many(
+    name,
+    scope = nil,
+    after_add: nil,
+    after_remove: nil,
+    association_foreign_key: nil,
+    autosave: nil,
+    before_add: nil,
+    before_remove: nil,
+    class_name: nil,
+    extend: nil,
+    foreign_key: nil,
+    join_table: nil,
+    validate: nil,
+    &blk
+  ); end
+end
+
+module ActiveRecord::NestedAttributes::ClassMethods
+  sig do
+    params(
+      attr_names: T.any(T.any(Symbol, String), T::Array[T.any(Symbol, String)]),
+      allow_destroy: T.nilable(T::Boolean),
+      reject_if: T.any(Symbol, T.proc.returns(T::Boolean)),
+      limit: T.any(Integer, Symbol, T.proc.returns(Integer)),
+      update_only: T.nilable(T::Boolean),
+    ).void
+  end
+  def accepts_nested_attributes_for(
+    attr_names,
+    allow_destroy: nil,
+    reject_if: nil,
+    limit: nil,
+    update_only: nil
+  ); end
+end
+
+module ActiveRecord::Scoping::Named::ClassMethods
+  sig do
+    params(
+      name: T.nilable(T.any(Symbol, String)),
+      body: T.nilable(T.any(
+        T.proc.returns(T.untyped),
+        T.proc.params(args: T.nilable(T.untyped)).returns(T.untyped)
+      )),
+      blk: T.nilable(T.proc.params(args: T.nilable(T.untyped)).returns(T.untyped))
+    ).void
+  end
+  def scope(
+    name,
+    body = nil,
```

Comment:
This is non-nullable 

https://github.com/rails/rails/blob/98a57aa5f610bc66af31af409c72173cdeeb3c9e/activerecord/lib/active_record/scoping/named.rb#L169

Reply:
Thank you!